security/pkg/pki/util/keycertbundle_test.go

```diff
@@ -343,145 +343,131 @@ func TestNewVerifiedKeyCertBundleFromFile(t *testing.T) {
 
 // Test the root cert expiry timestamp can be extracted correctly.
 func TestExtractRootCertExpiryTimestamp(t *testing.T) {
-	t0 := time.Now()
-	cert, key, err := GenCertKeyFromOptions(CertOptions{
-		Host:         "citadel.testing.istio.io",
-		NotBefore:    t0,
-		TTL:          time.Minute,
-		Org:          "MyOrg",
-		IsCA:         true,
-		IsSelfSigned: true,
-		IsServer:     true,
-		RSAKeySize:   2048,
-	})
-	if err != nil {
-		t.Errorf("failed to gen cert for Citadel self signed cert %v", err)
-	}
-	kb, err := NewVerifiedKeyCertBundleFromPem(cert, key, nil, cert)
-	if err != nil {
-		t.Errorf("failed to create key cert bundle: %v", err)
-	}
 	testCases := []struct {
-		name string
-		ttl  float64
-		time time.Time
+		NotBefore time.Time
```

Comment:
These tests weren't great because it would call `ExtractRootCertExpiryTimestamp` on the same keybundle in every test case.